src/osd/PrimaryLogPG.cc

```diff
@@ -7019,10 +7081,6 @@ int PrimaryLogPG::do_osd_ops(OpContext *ctx, vector<OSDOp>& ops)
 	  }
 	}
 
-        if (!oi.manifest.is_chunked()) {
-          oi.manifest.clear();
-        }
```

Comment:
This is not related to the patch. I just removed this because it is meaningless (SET_CHUNK case returns -EINVAL if the object is not set-chunk at upper part of SET_CHUNK case).  So, I am not sure which patch is right place to add this.

Also, we have discussed about this before in manifest.rst
Do I misunderstand you intention?

> TODO: SET_CHUNK appears to clear the manifest unconditionally if it's not chunked.
if (!oi.manifest.is_chunked()) {
  oi.manifest.clear();
}

Reply:
I'm not normally this careful about breaking up commits, but I've had a hard time understanding your PRs and I consistently find problems when I read in detail.  As such, I can't really just ignore something that doesn't seem to fit.  This process will be faster if you take care to only include in a commit changes related to that commit and break out unrelated changes to their own commits.

Please break this out to its own commit.

Reply:
Also, I think the context of my original comment was that if there was a non-chunked manifest (a redirect), we probably want to return an error rather than clear it.  Looks to me like further up in this portion we already do return an error if it's a redirect:

```
	if (oi.manifest.is_redirect()) {
	  result = -EINVAL;
	  goto fail;
	}
```

As such, I think this is actually just dead code.  Move it to its own commit with a commit message indicating that its no longer useful since we return an error in that case.

Reply:
got it